### ISSUE TITLE
Segfault on player join

### ISSUE THREAD

**Issue by Soren90** [splewis/csgo-pug-setup#368]
### Expected behavior
The server not to segfault on player join

### Actual behavior
Log from server (then server restart):
```
csgo-ds | L 01/27/2021 - 19:08:11: World triggered "Match_Start" on "de_dust2"
csgo-ds | /home/steam/csgo/srcds_run: line 318:   147 Segmentation fault      (core dumped) $HL_CMD
csgo-ds | Add "-debug" to the /home/steam/csgo/srcds_run command line to generate a debug.log to help with solving this problem
```
When running in -debug mode, the debug.log is empty

### Steps to reproduce

- Ubuntu 20.04
- Docker image: [kaimellea/csgo](https://github.com/kaimallea/csgo) (modified with the tar's in step 2:)
- Plugin version: 2.0.5
- Sourcemod version: sourcemod-1.10.0-git6502-linux.tar.gz
- Steps to reproduce (please be specific):
1. csgo clean install
2. installing these files:
https://mms.alliedmods.net/mmsdrop/1.11/mmsource-1.11.0-git1144-linux.tar.gz
https://sm.alliedmods.net/smdrop/1.10/sourcemod-1.10.0-git6502-linux.tar.gz
https://users.alliedmods.net/~kyles/builds/SteamWorks/SteamWorks-git132-linux.tar.gz
https://forums.alliedmods.net/showthread.php?t=169095 (updater.smx)
https://github.com/splewis/csgo-pug-setup/releases/download/2.0.5/pugsetup_2.0.5.zip
3.  setting this in pugsetup.cfg:
sm_pugsetup_autosetup 1 (makes it so users don't have to .setup to begin a pug)
sm_pugsetup_pausing_enabled 0 (disables pausing)
sm_pugsetup_force_defaults 1
4. Start server and join. (and then seg faults)
  
setting sm_pugsetup_autosetup 0 and setting it to 1 with rcon seems to work fine.
  
Am i missing some kind of dependency in the container?

**Comment by Soren90** [comment]
I also just tested moving all the plugins except pugsetup.smx to the disabled folder. And the issue is still there.

**Comment by r4f4w** [comment]
Same problem here!

What is funny is that i already have one server running this plugin, and when i grab the plugin that is totally functioning and throw into another fresh install of sourcemod, i get this anonnying error. (With same extensions, config cfg files, etc)

The working server is running SourceMod 1.10.0.6501, and the faulty one is running SourceMod 1.10.0.6502 (wich i don't think is gonna change a lot to give such an error)

**Comment by r4f4w** [comment]
I found it!!

The file that is conflicting with pugsetup is the last version of PTaH, located in:
`csgo/addons/sourcemod/gamedata/PTaH.games.txt`

I changed this file and it started to work. Wich is funny because PTaH is actually used by the **weapons** plugin.

**Comment by r4f4w** [comment]
Yup, confirmed, SourceMod 1.10.0.6501 is good but SourceMod 1.10.0.6502 is giving segmentation error.

Just go back one version and you're good to go.

**Comment by r4f4w** [comment]
It worked, then i close it, re-open and there it goes, the same error again. Can't figure out what is going on.

**Comment by r4f4w** [comment]
Now i changed to version 6499 and it's working

**Comment by r4f4w** [comment]
It seems like there is nothing to do with Sourcemod version

**Comment by r4f4w** [comment]
Yes, finally i got it!!

I'm running this server on docker image, so what i did was to give more cpu limit to my server. After this, i closed and open it again just to make sure, and it's working for sure now!

**Comment by Soren90** [comment]
Im my experience it seems like it has something to do with the server waking up from hibernation. I don't have any server availabe to test on at the moment, but the error was kind of inconsistent and but I was able to replicate the issue when I was the first player to join.

I haven't accually tried turning off hibernation. That could be a possible workaround

**Comment by Soren90** [comment]
> Yes, finally i got it!!
> 
> I'm running this server on docker image, so what i did was change my cpu pinning and give more cpu limit to my server. After this, i closed and open it again just to make sure, and it's working for sure now!

Hmm are you sure this solved the issue? I don't have any CPU limit on my server.

**Comment by r4f4w** [comment]
> > Yes, finally i got it!!
> > I'm running this server on docker image, so what i did was change my cpu pinning and give more cpu limit to my server. After this, i closed and open it again just to make sure, and it's working for sure now!
> 
> Hmm are you sure this solved the issue? I don't have any CPU limit on my server.

Yeah, i restarted 4 times, on the 5th time, the error goes again.

What makes me angry is that is not consistent, sometimes it works, and suddenly don't

**Comment by r4f4w** [comment]
> Im my experience it seems like it has something to do with the server waking up from hibernation. I don't have any server availabe to test on at the moment, but the error was kind of inconsistent and but I was able to replicate the issue when I was the first player to join.
> 
> I haven't accually tried turning off hibernation. That could be a possible workaround

So, this is actually working, but i won't confirm until i re-test like 30 times just to make sure 😂

Running more tests and i will give a report.

**Comment by r4f4w** [comment]
I'm having a little problem that my cfg is setting sv_hibernate_ms to 0, but when server start and i check, the value is default 20.

Don't know why my server.cfg is not changing that value.

Maybe because it has sm_cvar in front?

**Comment by r4f4w** [comment]
> I'm having a little problem that my cfg is setting sv_hibernate_ms to 0, but when server start and i check, the value is default 20.
> 
> Don't know why my server.cfg is not changing that value.
> 
> Maybe because it has sm_cvar in front?

Yup, removing sm_cvar got it to work, and now it's not giving more segmentation fault.

I will let you know if does come back, but as for now, thanks for the tip @Soren90 !

**Comment by mashsajid** [comment]
@Soren90 Did you fixed it?

**Comment by Soren90** [comment]
> @Soren90 Did you fixed it?

Nope. Still same issue. It seems like only workaround that works is turning off hibernation. Haven't tried it myself tho.

**Comment by r4f4w** [comment]
> > @Soren90 Did you fixed it?
> 
> Nope. Still same issue. It seems like only workaround that works is turning off hibernation. Haven't tried it myself tho.

I can confirm that turning off hibernation does work and solves the problem.

**Comment by mashsajid** [comment]
I had the same problem. It will be fixed by changing sm_pugsetup_use_game_warmup 1 to 0 in pugsetup.cfg

**Comment by rkantos** [comment]
I seem to be getting this with Get5 plugin. It doesn't seem to have the same kind of configuration for the warmup. I think also turning of hibernation is a workaround for this issue - I don't remember seeing the same issue with someone being on the server when it is configured by the addon+plugin.

**Comment by NaathySz** [comment]
I know this thing is old, but still happen and maybe i can help someone in future;
If you dont want turn this cvars talked above, you can do it:

try not leave server in LIVE, because it can cause this bug. Basically, if the match started and players leave in mid of match, when someone try join after (server hibernation + pug setup in live), it can cause this issue and crash. Maybe its not the only reason to this bug, but it can help anyway.

I know it cannot be our fault, so if you are in this trouble, just make this:

1. Restart your server and **NOT** join yet!!
2. Write **sm plugins unload pugsetup** in your server panel.
3. After plugin unload, try join. Plugin will load again automatically after you join and prevens to crash.

For me it work, i hope i can help someone with this.
And sorry my bad english :P

**Comment by Soren90** [comment]
> I know this thing is old, but still happen and maybe i can help someone in future; If you dont want turn this cvars talked above, you can do it:
> 
> try not leave server in LIVE, because it can cause this bug. Basically, if the match started and players leave in mid of match, when someone try join after (server hibernation + pug setup in live), it can cause this issue and crash. Maybe its not the only reason to this bug, but it can help anyway.
> 
> I know it cannot be our fault, so if you are in this trouble, just make this:
> 
> 1. Restart your server and **NOT** join yet!!
> 2. Write **sm plugins unload pugsetup** in your server panel.
> 3. After plugin unload, try join. Plugin will load again automatically after you join and prevens to crash.
> 
> For me it work, i hope i can help someone with this. And sorry my bad english :P

Thanks for reply! Have this worked long term? I the problem might show up again if all players leave the server and joins again after hibernation. But if you dont need that because you restart the server every time, then you should be fine.

Maybe a solution would be to set sm plugins unload pugsetup in server.cfg or somewhere else?

**Comment by NaathySz** [comment]
> > I know this thing is old, but still happen and maybe i can help someone in future; If you dont want turn this cvars talked above, you can do it:
> > try not leave server in LIVE, because it can cause this bug. Basically, if the match started and players leave in mid of match, when someone try join after (server hibernation + pug setup in live), it can cause this issue and crash. Maybe its not the only reason to this bug, but it can help anyway.
> > I know it cannot be our fault, so if you are in this trouble, just make this:
> > 
> > 1. Restart your server and **NOT** join yet!!
> > 2. Write **sm plugins unload pugsetup** in your server panel.
> > 3. After plugin unload, try join. Plugin will load again automatically after you join and prevens to crash.
> > 
> > For me it work, i hope i can help someone with this. And sorry my bad english :P
> 
> Thanks for reply! Have this worked long term? I the problem might show up again if all players leave the server and joins again after hibernation. But if you dont need that because you restart the server every time, then you should be fine.
> 
> Maybe a solution would be to set sm plugins unload pugsetup in server.cfg or somewhere else?

it will not solve, but i will talk to splewis about it, lets see if he fix.
If not, i can try do it myself

**Comment by splewis** [comment]
I don't have any magic knowledge on why the server crashes here, or how to fix it. It's exceptionally hard to debug server crashes since valve removed debug symbols from the server binary; there's not much I can do to help. "Fixing" such a bug is likely just trying random ideas and trial+error until things "work". If someone actually uploads crash dumps from [accelerator](https://forums.alliedmods.net/showthread.php?t=277703) that might give some hints as to where the crash happens.

In general, hibernation can be somewhat buggy when it comes to plugin interactions, so I would recommend it be turned off.

**Comment by NaathySz** [comment]
[pugsetup_2.0.8.zip](https://github.com/splewis/csgo-pug-setup/files/9355010/pugsetup_2.0.8.zip)

Here is a fix. It will stop match if its live and only 1 player last in server. He will no play alone in LIVE match, most just disconnect too. So, it will back to warmup and auto setup again.

Better than crash, huh?

**Comment by NaathySz** [comment]
[pugsetup_2.0.8.zip](https://github.com/splewis/csgo-pug-setup/files/9355136/pugsetup_2.0.8.zip)
